src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,216 @@
+            normalize_score=False,
+            n_resample_complement_sets=n_resample_complement_sets,
+            progress=progress,
+        ),
+        n_jobs=n_jobs,
+        config=config,
+    )
+    result = map_reduce_job()
+
+    if normalize_score:
+        _normalize_classwise_shapley_values(result, u)
+
+    return result
+
+
+def _classwise_shapley_worker(
+    update_indices: NDArray[np.int_],
+    u: Utility,
+    *,
+    done: StoppingCriterion,
+    truncation: TruncationPolicy,
+    normalize_score: bool = False,
+    n_resample_complement_sets: int = 1,
+    progress: bool = True,
+) -> ValuationResult:
+    """Computes class-wise shapley value by using a truncated monte-carlo permutation
+    samples for the subsets.
+
+    :param update_indices: Subset of all data points in ``u`` to be valued.
+    :param u: Utility object with model, data, and scoring function. The scoring
+        function has to be of type :class:`~pydvl.utils.score.ClassWiseScorer`.
+    :param done: Function checking whether computation must stop.
+    :param truncation: A callable which decides whether to interrupt processing a
+        permutation and set all subsequent marginals to zero.
+    :param normalize_score: Whether to normalize the score by the number of classes.
+        This is handy to make the values comparable across different runs.
+    :param n_resample_complement_sets: How often the complement set shall be resampled
+        for each permutation.
+    :param progress: Whether to display progress bars for each job.
+    :return: ValuationResult object with the data values.
+    """
+    _check_classwise_shapley_utility(u)
+
+    result = ValuationResult.zeros(
+        algorithm="classwise_shapley",
+    )
+
+    x_train, y_train = u.data.get_training_data(update_indices)
+    unique_labels = np.unique(y_train)
+    pbar = tqdm(disable=not progress, position=0, total=100, unit="%")
+    scorer = cast(ClassWiseScorer, u.scorer)
+
+    for n_step in itertools.count():
+
+        pbar.n = 100 * done.completion()
+        pbar.refresh()
+
+        for label in unique_labels:
+            logger.debug(
+                f"Sampling subset #{n_step} complement index set of class '{label}'."
+            )
+            scorer.label = label
+            result += permutation_montecarlo_classwise_shapley(
+                u,
+                label,
+                done=MaxChecks(n_resample_complement_sets),
+                truncation=truncation,
+            )
+
+        if done(result):
+            break
+
+    if normalize_score:
+        _normalize_classwise_shapley_values(result, u)
+
+    return result
+
+
+def _check_classwise_shapley_utility(u: Utility):
+    """
+    Checks whether the passed utility object is suitable for computing the class-wise
+    shapley values. Therefore, it is checked that the data passed is a pure
+    classification dataset. If the function passes without check it is considered to be
+    run successful.
```

Comment:
English fix and simplification:
```suggestion
    Verifies if the provided utility object can calculate class-wise Shapley values.
    This function ensures the input data is strictly for classification purposes. If the
    test is successful, the function runs without issues.
```

The text above is the output of GPT-4 with the prompt:
> Please fix the English usage and simplify as much as possible this text, using language suitable for software documentation